examples/deserialize-transaction/src/example.ts

```diff
@@ -0,0 +1,403 @@
+// Let's verify that it's correct
+const signedByAddress = 'ED1WqT2hWJLSZtj4TtTdoovmpMrr7zpkUdbfxmcJR1Fq' as Address;
+const signedBySignature = decodedTransaction.signatures[signedByAddress]!;
+// We encode the source address to bytes
+const sourceAddressBytes = getAddressEncoder().encode(signedByAddress);
+// Then we create a public Ed25519 key with those bytes
+const signedByPublicKey = await crypto.subtle.importKey('raw', sourceAddressBytes, 'Ed25519', true, ['verify']);
+// Now we can verify the signature using that key
+const verifiedSignature = await verifySignature(signedByPublicKey, signedBySignature, decodedTransaction.messageBytes);
+log.info(
+    `[step 1] The signature for ${signedByAddress} is ${verifiedSignature ? 'valid' : 'not valid'} for the transaction`,
+);
+
+/**
+ * STEP 2: DECODE TO COMPILED TRANSACTION MESSAGE
+ * We verified the signature is valid for the `messageBytes` field of the Transaction. This is a byte array
+ * representing the compiled version of the `TransactionMessage`. If we decode this we can see some information
+ * about the transaction. Let's do that next.
+ */
+
+// Again we have a decoder to convert from bytes to a Solana data structure, in this case the `CompiledTransactionMessage`
+const compiledTransactionMessageDecoder = getCompiledTransactionMessageDecoder();
+const compiledTransactionMessage = compiledTransactionMessageDecoder.decode(decodedTransaction.messageBytes);
+
+// This gives us the data stucture `CompiledTransactionMessage`. This is the format that transactions are
+// compiled to before being sent to the Solana network.
+
+// Let's inspect some fields of `compiledTransactionMessage`
+// We can see its version:
+log.info(`[step 2] The transaction is version ${compiledTransactionMessage.version}`);
+
+// We can see the lifetime token, though we don't have enough context yet to know if it's a blockhash or a durable nonce
+log.info(
+    `[step 2] We can see the transaction lifetime token, but we don't know if it's a blockhash or durable nonce: ${compiledTransactionMessage.lifetimeToken}`,
+);
+
+// We can see the static accounts:
+log.info(compiledTransactionMessage.staticAccounts, '[step 2] Static accounts of the transaction');
+
+// Note that the destination address (F1Vc6AGoxXLwGB7QV8f4So3C5d8SXEk3KKGHxKGEJ8qn) is not here, as it comes from the lookup table
+
+// The `addressLookupTables` field is only included for non-legacy transactions
+// Here we tell Typescript to narrow the type to exclude legacy transactions
+if (compiledTransactionMessage.version === 'legacy') {
+    throw new Error('We used version: 0');
+}
+
+// Now we can view address lookup tables:
+log.info(compiledTransactionMessage.addressTableLookups, '[step 2] Address lookup tables for the transaction');
+
+// We see that the transaction uses our address lookup table.
+
+// Let's look at a compiled instruction
+log.info(
+    { ...compiledTransactionMessage.instructions[0], data: '(removed for brevity)' },
```

Comment:
nit: Since this is a transfer, the instruction data should only be 8 bytes, right?

Reply:
I'm getting 12 bytes, I guess it's using a 4 byte discriminator

But the reason I did this is that the logger we use unrolls the array to one element per line, like this:

```
INFO (Deserialize): [step 2] The first instruction of the compiled transaction message
    programAddressIndex: 2
    accountIndices: [
      1,
      4
    ]
    data: {
      "0": 2,
      "1": 0,
      "2": 0,
      "3": 0,
      "4": 78,
      "5": 97,
      "6": 188,
      "7": 0,
      "8": 0,
      "9": 0,
      "10": 0,
      "11": 0
    }
```

Which I don't think is useful in this case until we actually want to look at the data later on